src/pages/api/post-review.ts

```diff
@@ -0,0 +1,48 @@
+// pages/api/createService.ts
+import { NextApiRequest, NextApiResponse } from 'next';
+import { Contract, ethers, Wallet } from 'ethers';
+import { config } from '../../config';
+import TalentLayerReview from '../../contracts/ABI/TalentLayerReview.json';
+import { getUserByAddress } from '../../queries/users';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { userId, userAddress, serviceId, uri, valuesRating } = req.body;
+  let signer;
+
+  // @dev : you can add here all the check you need to confirm the delagation for a user
+
+  try {
+    if (process.env.NEXT_PUBLIC_ACTIVE_DELEGATE !== 'true') {
```

Comment:
from this line to line 34 it's the same for all posts, so to prevent duplicates move to a utilis.ts file with a function called getDelegationSigner in the delegate folder which take res as arg and return the signer if everything is good